installer/scripts/omsadmin.sh

```diff
@@ -514,17 +538,34 @@ onboard()
         --output "$RESP_ONBOARD" $CURL_VERBOSE \
         --write-out "%{http_code}\n" $PROXY_SETTING \
         https://${WORKSPACE_ID}.oms.${URL_TLD}/AgentService.svc/LinuxAgentTopologyRequest` || error=$?
-                 
-    if [ $error -ne 0 ]; then
+
+    if [ $error -eq $CURL_HOST_RESOLVE_ERROR -a -z "$PROXY_SETTING" ]; then
+        log_error "Error resolving host during the onboarding request. Check the correctness of the workspace ID and the internet connectivity, or add a proxy."
+        cleanup_certs
+        return $ERROR_RESOLVING_HOST
+    elif [ $error -eq $CURL_PROXY_RESOLVE_ERROR -a -n "$PROXY_SETTING" ]; then
+        log_error "Proxy could not be resolved during the onboarding request. Verify the proxy."
+        cleanup_certs
+        return $INVALID_PROXY
+    elif [ $error -eq $CURL_CONNECT_HOST_ERROR -a -n "$PROXY_SETTING" ]; then
+        log_error "Error connecting to OMS service through proxy. Verify the proxy."
+        cleanup_certs
+        return $INVALID_PROXY
+    elif [ $error -ne 0 ]; then
         log_error "Error during the onboarding request. Check the correctness of the workspace ID and shared key or run omsadmin.sh with '-v'"
-        rm "$FILE_CRT" "$FILE_KEY" > /dev/null 2>&1 || true
-        return 1
+        cleanup_certs
+        return $ERROR_ONBOARDING
     fi
 
     if [ "$RET_CODE" = "200" ]; then
         $RUBY $MAINTENANCE_TASKS_SCRIPT --endpoints "$RESP_ONBOARD","$ENDPOINT_FILE" "$CONF_OMSADMIN" "$FILE_CRT" "$FILE_KEY" "$RUN_DIR/omsagent.pid" "$CONF_PROXY" "$OS_INFO" "$INSTALL_INFO" $CURL_VERBOSE
-        if [ $? -ne 0 ]; then
-            log_warning "During onboarding request, certificate update and DSC endpoints may not have been extracted."
+        endpoints_ret=$?
```

Comment:
The agent maintenance script returns 12 different error codes, but only three are exposed here. If the rest are not exposed, why was the main script modified to return them?